lxd-agent/main_agent.go

```diff
@@ -176,6 +166,16 @@ func (c *cmdAgent) Run(cmd *cobra.Command, args []string) error {
 		}
 	}()
 
+	// Done with early setup, tell systemd to continue boot.
+	// Allows a service that needs a file that's generated by the agent to be able to declare After=lxd-agent
+	// and know the file will have been created by the time the service is started.
+	if os.Getenv("NOTIFY_SOCKET") != "" {
```

Comment:
+1 on cloud-init side which experiences a race on Focal VMs when systemd-notify allows cloud-init-local.service to start but sometimes /dev/lxc/sock wasn't present yet.